src/infra/core/src/pull_service_impl.rs

```diff
@@ -139,7 +139,7 @@ impl PullServiceImpl {
                 Some(
                     DatasetRefRemote::Alias(alias)
                     | DatasetRefRemote::Handle(DatasetHandleRemote { alias, .. }),
-                ) => DatasetAlias::new(alias.account_name.clone(), alias.dataset_name.clone()),
+                ) => DatasetAlias::new(None, alias.dataset_name.clone()),
```

Comment:
`kamu pull repo/account/dataset` was inferring destination alias as `account/dataset` which was of course confusing the single-tenant repo (panic). In case of a multi-tenant local repo I think it's still correct to default to current user (account=`None`)